components/Weather/index.js

```diff
@@ -0,0 +1,76 @@
+import axios from "axios";
+import { useState, useEffect } from "react";
+import styled from "styled-components";
+import Icon from "../Icons";
+
+export default function Weather({ outivity }) {
+  const [weatherData, setWeatherData] = useState(null);
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  const fetchData = async () => {
+    try {
+      const response = await axios.get(
+        `https://api.openweathermap.org/data/2.5/weather?lat=${outivity.lat}&lon=${outivity.lng}&units=metric&appid=9e7a02e3cc30d92b7c137c9919b52688`
+      );
+
+      setWeatherData(response.data);
+      setErrorMessage(null);
+    } catch (error) {
+      console.error("Error fetching weather data: ", error);
+      setErrorMessage("Weather data is not available at the moment...");
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, [outivity.lat, outivity.lng]);
+
+  return (
+    <>
+      {errorMessage ? (
+        <StyledInfoMessage>{errorMessage}</StyledInfoMessage>
+      ) : weatherData ? (
+        <StyledWeatherContainer>
+          <StyledWeatherInfo>
+            <Icon variant={"temperature"} size={25} />
+            <StyledWeatherData>{weatherData.main.temp}°C</StyledWeatherData>
+          </StyledWeatherInfo>
+          <StyledWeatherInfo>
+            <Icon variant={"weather"} size={25} />
+            <StyledWeatherData>
+              {weatherData.weather[0].description}
+            </StyledWeatherData>
+          </StyledWeatherInfo>
+          <StyledWeatherInfo>
+            <Icon variant={"humidity"} size={25} />
+            <StyledWeatherData>{weatherData.main.humidity}%</StyledWeatherData>
+          </StyledWeatherInfo>
+        </StyledWeatherContainer>
+      ) : (
+        <StyledInfoMessage>Loading weather data...</StyledInfoMessage>
+      )}
+    </>
```

Comment:
Nested ternary operators are hard to read. It's okay but if you can structure the code more readable it would be nice.

Reply:
We would keep it this way for now and then rewrite the code in the future brunches